README.md

```diff
@@ -19,81 +19,265 @@
-| `tensorflow_model.keras`     | TensorFlow         | Keras format (default)          |
-| `tensorflow_hdf5_model.h5`   | TensorFlow         | Legacy HDF5 format              |
-| `tensorflow_hdf5.weights.h5` | TensorFlow         | Legacy HDF5 weights only format |
-| `pytorch_model.pth`          | PyTorch            | PyTorch default format          |
-| `pytorch_full_model.pth`     | PyTorch            | PyTorch complete model format   |
-| `pytorch_jitted_model.pt`    | PyTorch            | PyTorch TorchScript format      |
+The CLI has two subcommands: `sign` for signing and `verify` for verification.
+Each subcommand has another level of subcommands to select the signing method
+(`sigstore` -- the default, can be skipped --, `key`, `certificate`). Then, each
+of these subcommands has several flags to configure parameters for
+signing/verification.
 
-See [slsa_for_models/README.md](slsa_for_models/README.md) for more information.
+For the demo, we will use the `bert-base-uncased` model, which can be obtained
+via:
+
+```bash
+[...]$ git clone --depth=1 "https://huggingface.co/bert-based-uncased
+```
+
+We remove the `.git` directory since that should not be included in the
+signature:
+
+```bash
+[...]$ rm -rf bert-base-uncased/.git
+```
+
+By default, the code also ignores git related paths.
+
+The simplest example of the CLI is to sign a model using Sigstore:
+
+```bash
+[...]$ model_signing sign bert-base-uncased
+```
+
+This will open an OIDC flow to obtain a short lived token for the certificate.
+The identity used during signing and the provider must be reused during
+verification.
+
+As another example, here is how we can sign with private keys. First, we
+generate the key pair:
+
+```bash
+[...]$ openssl ecparam -name prime256v1 -genkey -noout -out key.priv
+[...]$ openssl ec -in key.priv -pubout > key.pub
+```
+
+And then we use the private key to sign.
+
+```bash
+[...]$ model_signing sign key bert-base-uncased --private_key key.priv
+```
+
+All signing methods support changing the signature name and location via the
+`--signature` flag:
+
+```bash
+[...]$ model_signing sign bert-base-uncased --signature model.sig
+```
+
+Consult the help for a list of all flags (`model_signing --help`, or directly
+`model_signing` with no arguments)
+
+On verification we use the `verify` subcommand. To verify a Sigstore signed
+model we use
+
+```bash
+[...]$ model_signing verify bert-base-uncased \
+      --signature model.sig \
+      --identity "$identity" \
+      --identity_provider "$oidc_provider"
+```
+
+Where `$identity` and `$oidc_provider` are those set up during the signing flow
+and `--signature` must point to the signature to verify.
+
+For developers signing models with Sigstore, there are three identity providers
+that can be used at the moment:
+
+* Google's provider is `https://accounts.google.com`.
+* GitHub's provider is `https://github.com/login/oauth`.
+* Microsoft's provider is `https://login.microsoftonline.com`.
```

Comment:
github actions has a different identity provider `https://token.actions.githubusercontent.com`

Reply:
Nice, added. Thank you